nomad/leader.go

```diff
@@ -1966,43 +1963,79 @@ func (s *Server) getOrCreateSchedulerConfig() *structs.SchedulerConfiguration {
 	return config
 }
 
+var minVersionKeyring = version.Must(version.NewVersion("1.4.0"))
+
 // initializeKeyring creates the first root key if the leader doesn't
 // already have one. The metadata will be replicated via raft and then
 // the followers will get the key material from their own key
 // replication.
-func (s *Server) initializeKeyring() error {
+func (s *Server) initializeKeyring(stopCh <-chan struct{}) {
+
+	logger := s.logger.Named("keyring")
 
 	store := s.fsm.State()
 	keyMeta, err := store.GetActiveRootKeyMeta(nil)
 	if err != nil {
-		return err
+		logger.Error("failed to get active key: %v", err)
```

Comment:
Is there anything else we can/should say here? If it's an ignorable part of an upgrade process let's drop it to warn.

Reply:
This is one of those annoying errors that bubbles up from the state store's `txn.Get` that can't actually ever fail unless we give `txn.Get` the wrong table name, which isn't ever going to happen in prod builds. We could probably eliminate a lot of spurious error checking if we panicked in the state store on those cases, but the current idiom in the state store code is to bubble it up.